server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -217,6 +217,49 @@ public CloseableIterator<DataSegmentPlus> retrieveUnusedSegmentsPlus(
     );
   }
 
+  public List<DataSegmentPlus> retrieveSegmentsById(String datasource, Set<String> segmentIds)
+  {
+    final List<List<String>> partitionedSegmentIds
+        = Lists.partition(new ArrayList<>(segmentIds), 100);
```

Comment:
We have a few batching constants that are set to 100: `MAX_NUM_SEGMENTS_TO_ANNOUNCE_AT_ONCE` and `MAX_INTERVALS_PER_BATCH`. Perhaps in a future patch, we can consolidate these to use a single constant since the intent is the same.

Reply:
Yeah, makes sense. Will do it in a follow up.